load_testing/common_flows.py

```diff
@@ -0,0 +1,154 @@
+
+        if not token:
+            resp.failure(
+                "Not a sign-in page. Current URL is {}.".format(resp.url)
+            )
+
+    resp = context.client.post(
+        resp.url,
+        data={
+            'user[email]': credentials['email'],
+            'user[password]': credentials['password'],
+            'authenticity_token': authenticity_token(dom),
+            'commit': 'Submit',
+        },
+        catch_response=True
+    )
+    dom = resp_to_dom(resp)
+    code = dom.find("#code").attr('value')
+
+    if not code:
+        resp.failure(
+            """
+            No 2FA code found.
+            Make sure {} is created and can log into the IDP
+            """.format(credentials)
+        )
+        return
+
+    code_form = dom.find("form[action='/login/two_factor/sms']")
+    
+    resp = context.client.post(
+        code_form.attr('action'),
+        data={
+            'code': code,
+            'authenticity_token': authenticity_token(dom),
+            'commit': 'Submit'
+        }
+    )
+
+    # We're not checking for post-login state here,
+    # as it will vary depending on the SP.
+    resp.raise_for_status()
+
+    return resp
+
+
+"""
+*** Sign Up Flow ***
+"""
+def do_sign_up(context):
+  fake = Faker()
+  phone_numbers = foney.phone_numbers()
+  default_password = "salty pickles"
+
+  # GET the new email page
+  resp = context.client.get("/sign_up/enter_email")
+  dom = resp_to_dom(resp)
+
+  # Post fake email and get confirmation link (link shows up in "load test mode")
+  new_email = "test+{}@test.com".format(fake.md5())
+  resp = context.client.post("/sign_up/enter_email",
+            data={
+              'user[email]': new_email,
+              'authenticity_token': authenticity_token(dom),
+            },
+            catch_response=True)
+
+  with resp as confirm_resp:
+      dom = resp_to_dom(confirm_resp)
+      try:
+          confirmation_link = dom.find("#confirm-now")[0].attrib['href']
+      except Exception as err:
+          resp.failure(
+              "Could not find CONFIRM NOW link, is IDP enable_load_testing_mode: 'true' ?" + err)
+          return 
+  
+  # Get confirmation token
+  resp = context.client.get(confirmation_link, name="/sign_up/email/confirm?confirmation_token=")
+  dom = resp_to_dom(resp)
+  token = dom.find('[name="confirmation_token"]:first').attr('value')
+
+  # Set user password
+  resp = context.client.post("/sign_up/create_password",
+            data={
+                'password_form[password]': default_password,
+                'authenticity_token': authenticity_token(dom),
+                'confirmation_token': token,
+            }
+        )
+
+  # After password creation set up SMS 2nd factor
+  phone_setup_url = "/phone_setup"
+  resp = context.client.get(phone_setup_url)
+  dom = resp_to_dom(resp)
+  auth_token = authenticity_token(dom)
+
+  resp = context.client.post(
+    phone_setup_url,
+    data={
+      '_method': 'patch',
```

Comment:
Looks like there's a `patch` method that we could use instead of `.post()`, should we give that a shot?

https://docs.locust.io/en/v0.7/api.html#locust.clients.HttpSession.patch

Reply:
💡

Reply:
Works great, thanks@